apps/web/src/components/TextInput/TextInput.js

```diff
@@ -36,21 +36,50 @@ const TextInput = forwardRef(({
   const clear = () => onChange && onChange({ target: { name, value: '' } })
 
   const [active, setActive] = useState(false)
+  const inputEl = useRef(null)
+
+  // Combine forwarded ref and local ref
+  useEffect(() => {
+    if (ref) {
+      if (typeof ref === 'function') {
+        ref(inputEl.current)
+      } else {
+        ref.current = inputEl.current
+      }
+    }
+  }, [ref])
+
+  // Check for autofill after mount
+  useEffect(() => {
+    const checkAutofill = () => {
```

Comment:
what does autofill mean here?  is this always focusing on the input every time? that doesnt seem right

Reply:
We want to check if autofill has happened every time the component mounts, in order to adjust the styles... Maybe you can think of a better way? Otherwise it defaults to a white background that looks weird. Or are you saying this is too expensive/inefficient?

Reply:
but all this is doing is setting the field to active, which doesnt make sense to me. Only one field should be active at a time, this will make all the autofilled fields active at the same time? shouldnt we use a different variable to indicate autofilled? also i see you using input:-webkit-autofill, below, that only works in safari not other browsers?